From c64865691df6d2ad18aeecaecb0cb9a94b5d71eb Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 19 Sep 2018 10:56:29 -0400
Subject: [PATCH] xwayland: Don't initialize glamor on llvmpipe

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
 hw/xwayland/xwayland-glamor-gbm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hw/xwayland/xwayland-glamor-gbm.c b/hw/xwayland/xwayland-glamor-gbm.c
index 06fcf52..2144af7 100644
--- a/hw/xwayland/xwayland-glamor-gbm.c
+++ b/hw/xwayland/xwayland-glamor-gbm.c
@@ -843,6 +843,11 @@ xwl_glamor_gbm_init_egl(struct xwl_screen *xwl_screen)
         goto error;
     }
 
+    if (strstr(glGetString(GL_RENDERER), "llvmpipe")) {
+        ErrorF("Refusing to try glamor with llvmpipe\n");
+        goto error;
+    }
+
     if (!epoxy_has_gl_extension("GL_OES_EGL_image")) {
         ErrorF("GL_OES_EGL_image not available\n");
         goto error;
-- 
2.17.1

